diff --git a/src/paths.js b/src/paths.js
index 1cc5ca7..d8c71aa 100644
--- a/src/paths.js
+++ b/src/paths.js
@@ -4,7 +4,7 @@ const { app } = require('electron');
 
 const buildInfo = require('./utils/buildInfo');
 
-let userData, userDataVersioned, resourcesPath, moduleData, exeDir, installPath;
+let userData, userDataVersioned, resourcesPath, moduleData, exeDir, installPath, logPath;
 
 exports.getUserData = () => userData;
 exports.getUserDataVersioned = () => userDataVersioned;
@@ -12,6 +12,7 @@ exports.getUserDataVersioned = () => userDataVersioned;
 exports.getResources = () => resourcesPath;
 exports.getModuleDataPath = () => moduleData;
 exports.getInstallPath = () => installPath;
+exports.getLogPath = () => logPath;
 
 exports.getExeDir = () => exeDir;
 
@@ -26,7 +27,9 @@ exports.init = () => {
 
   moduleData = buildInfo.newUpdater ? join(userData, 'module_data') : join(userDataVersioned, 'modules');
   resourcesPath = join(process.resourcesPath);
+  logPath = join(userData, 'logs');
 
   global.moduleDataPath = moduleData; // Global because discord
+  global.logPath = logPath; // Global because discord
   app.setPath('userData', userData); // Set userData properly because electron
-};
\ No newline at end of file
+};
