diff --git a/resources/assets/v1/src/components/transactions/EditTransaction.vue b/resources/assets/v1/src/components/transactions/EditTransaction.vue
index 97fbf78f41..20232584f6 100644
--- a/resources/assets/v1/src/components/transactions/EditTransaction.vue
+++ b/resources/assets/v1/src/components/transactions/EditTransaction.vue
@@ -470,7 +470,7 @@ export default {
             let result = {
                 transaction_journal_id: transaction.transaction_journal_id,
                 description: transaction.description,
-                date: transaction.date.substring(0, 16),
+                date: transaction.date.substring(0, 19),
                 reconciled: transaction.reconciled,
                 amount: this.roundNumber(this.positiveAmount(transaction.amount), transaction.currency_decimal_places),
                 category: transaction.category_name,
