diff --git a/config.def.h b/config.def.h
index 061ad66..2356e90 100644
--- a/config.def.h
+++ b/config.def.h
@@ -1,4 +1,5 @@
 /* See LICENSE file for copyright and license details. */
+#include <X11/XF86keysym.h>
 
 /* appearance */
 static const unsigned int borderpx  = 1;        /* border pixel of windows */
@@ -45,7 +46,7 @@ static const Layout layouts[] = {
 };
 
 /* key definitions */
-#define MODKEY Mod1Mask
+#define MODKEY Mod4Mask
 #define TAGKEYS(KEY,TAG) \
 	{ MODKEY,                       KEY,      view,           {.ui = 1 << TAG} }, \
 	{ MODKEY|ControlMask,           KEY,      toggleview,     {.ui = 1 << TAG} }, \
@@ -57,12 +58,27 @@ static const Layout layouts[] = {
 
 /* commands */
 static const char *dmenucmd[] = { "dmenu_run", "-fn", dmenufont, "-nb", col_gray1, "-nf", col_gray3, "-sb", col_cyan, "-sf", col_gray4, NULL };
-static const char *termcmd[]  = { "st", NULL };
+static const char *termcmd[]  = { "alacritty", NULL };
+static const char *screenshotcmd[] = { "flameshot", "gui", NULL };
+static const char *clipmenucmd[] = { "clipmenu", "-fn", dmenufont, "-nb", col_gray1, "-nf", col_gray3, "-sb", col_cyan, "-sf", col_gray4, NULL };
+static const char *raisevolumecmd[] = { "pulsemixer", "--change-volume", "+5", NULL };
+static const char *lowervolumecmd[] = { "pulsemixer", "--change-volume", "-5", NULL };
+static const char *muteoutputcmd[] = { "pulsemixer", "--toggle-mute", NULL };
+static const char *nextsongcmd[] = { "playerctl", "next", NULL };
+static const char *prevsongcmd[] = { "playerctl", "previous", NULL };
+static const char *playpausesongcmd[] = { "playerctl", "play-pause", NULL };
+static const char *mutemiccmd[] = { "sh", "-c", "pulsemixer --id $(pulsemixer --list-sources | grep Default | cut -d ' ' -f 3 | cut -d ',' -f 1) --toggle-mute", NULL };
+static const char *lockcmd[] = { "slock", NULL };
+static const char *tgspdunstcmd[] = { "toggle-dunst-notifications", NULL };
+static const char *incbrightcmd[] = { "light", "-A", "5", NULL };
+static const char *decbrightcmd[] = { "light", "-U", "5", NULL };
 
 static const Key keys[] = {
 	/* modifier                     key        function        argument */
-	{ MODKEY,                       XK_p,      spawn,          {.v = dmenucmd } },
-	{ MODKEY|ShiftMask,             XK_Return, spawn,          {.v = termcmd } },
+	{ MODKEY,                       XK_e,      spawn,          {.v = dmenucmd } },
+	{ MODKEY,                       XK_Return, spawn,          {.v = termcmd } },
+	{ 0,                            XK_Print,  spawn,          {.v = screenshotcmd } },
+	{ MODKEY,                       XK_v,      spawn,          {.v = clipmenucmd } },
 	{ MODKEY,                       XK_b,      togglebar,      {0} },
 	{ MODKEY,                       XK_j,      focusstack,     {.i = +1 } },
 	{ MODKEY,                       XK_k,      focusstack,     {.i = -1 } },
@@ -70,9 +86,9 @@ static const Key keys[] = {
 	{ MODKEY,                       XK_d,      incnmaster,     {.i = -1 } },
 	{ MODKEY,                       XK_h,      setmfact,       {.f = -0.05} },
 	{ MODKEY,                       XK_l,      setmfact,       {.f = +0.05} },
-	{ MODKEY,                       XK_Return, zoom,           {0} },
+	{ MODKEY|ShiftMask,             XK_Return, zoom,           {0} },
 	{ MODKEY,                       XK_Tab,    view,           {0} },
-	{ MODKEY|ShiftMask,             XK_c,      killclient,     {0} },
+	{ MODKEY,                       XK_q,      killclient,     {0} },
 	{ MODKEY,                       XK_t,      setlayout,      {.v = &layouts[0]} },
 	{ MODKEY,                       XK_f,      setlayout,      {.v = &layouts[1]} },
 	{ MODKEY,                       XK_m,      setlayout,      {.v = &layouts[2]} },
@@ -80,10 +96,11 @@ static const Key keys[] = {
 	{ MODKEY|ShiftMask,             XK_space,  togglefloating, {0} },
 	{ MODKEY,                       XK_0,      view,           {.ui = ~0 } },
 	{ MODKEY|ShiftMask,             XK_0,      tag,            {.ui = ~0 } },
-	{ MODKEY,                       XK_comma,  focusmon,       {.i = -1 } },
-	{ MODKEY,                       XK_period, focusmon,       {.i = +1 } },
-	{ MODKEY|ShiftMask,             XK_comma,  tagmon,         {.i = -1 } },
-	{ MODKEY|ShiftMask,             XK_period, tagmon,         {.i = +1 } },
+	{ MODKEY,                       XK_comma,  focusmon,       {.i = +1 } },
+	{ MODKEY,                       XK_period, focusmon,       {.i = -1 } },
+	{ MODKEY|ShiftMask,             XK_comma,  tagmon,         {.i = +1 } },
+	{ MODKEY|ShiftMask,             XK_period, tagmon,         {.i = -1 } },
+	{ MODKEY,                       XK_o,      spawn,          {.v = lockcmd } },
 	TAGKEYS(                        XK_1,                      0)
 	TAGKEYS(                        XK_2,                      1)
 	TAGKEYS(                        XK_3,                      2)
@@ -93,9 +110,23 @@ static const Key keys[] = {
 	TAGKEYS(                        XK_7,                      6)
 	TAGKEYS(                        XK_8,                      7)
 	TAGKEYS(                        XK_9,                      8)
-	{ MODKEY|ShiftMask,             XK_q,      quit,           {0} },
+	{ MODKEY|ControlMask,           XK_q,      quit,           {0} },
+	/* MEDIA KEYS */
+	{ 0,                            XF86XK_AudioLowerVolume,     spawn, {.v = lowervolumecmd } },
+	{ 0,                            XF86XK_AudioMute,            spawn, {.v = muteoutputcmd } },
+	{ 0,                            XF86XK_AudioRaiseVolume,     spawn, {.v = raisevolumecmd } },
+	{ 0,                            XF86XK_AudioNext,            spawn, {.v = nextsongcmd } },
+	{ 0,                            XF86XK_AudioPrev,            spawn, {.v = prevsongcmd } },
+	{ 0,                            XF86XK_AudioPlay,            spawn, {.v = playpausesongcmd } },
+	{ 0,                            XK_Pause,                    spawn, {.v = mutemiccmd } },
+	{ 0,                            XF86XK_PowerOff,             spawn, {.v = mutemiccmd } },
+	{ 0,                            XF86XK_MonBrightnessUp,      spawn, {.v = incbrightcmd } },
+	{ 0,                            XF86XK_MonBrightnessDown,    spawn, {.v = decbrightcmd } },
+	/* OTHER ACTIONS */
+	{ MODKEY,                       XK_n,      spawn,          {.v = tgspdunstcmd } },
 };
 
+
 /* button definitions */
 /* click can be ClkTagBar, ClkLtSymbol, ClkStatusText, ClkWinTitle, ClkClientWin, or ClkRootWin */
 static const Button buttons[] = {
