diff --git a/config.def.h b/config.def.h
index 2356e90..f18df8d 100644
--- a/config.def.h
+++ b/config.def.h
@@ -6,7 +6,11 @@ static const unsigned int borderpx  = 1;        /* border pixel of windows */
 static const unsigned int snap      = 32;       /* snap pixel */
 static const int showbar            = 1;        /* 0 means no bar */
 static const int topbar             = 1;        /* 0 means bottom bar */
-static const char *fonts[]          = { "monospace:size=10" };
+static const char *fonts[]          = {
+										"Material Design Icons:size=10",
+										"Fira Code:size=10",
+										"monospace:size=10"
+										};
 static const char dmenufont[]       = "monospace:size=10";
 static const char col_gray1[]       = "#222222";
 static const char col_gray2[]       = "#444444";

