diff --git a/config.def.h b/config.def.h
index 3a79fa5..93d3c36 100644
--- a/config.def.h
+++ b/config.def.h
@@ -20,14 +20,27 @@ static const char col_gray2[]       = "#444444";
 static const char col_gray3[]       = "#bbbbbb";
 static const char col_gray4[]       = "#eeeeee";
 static const char col_cyan[]        = "#005577";
+static const char col_red[]         = "#BF616A";
+static const char col_orange[]      = "#D08770";
+static const char col_yellow[]      = "#EBCB8B";
+static const char col_green[]       = "#A3BE8C";
+static const char col_pink[]        = "#B48EAD";
+
 static const char *colors[][3]      = {
 	/*               fg         bg         border   */
 	[SchemeNorm] = { col_gray3, col_gray1, col_gray2 },
 	[SchemeSel]  = { col_gray4, col_cyan,  col_cyan  },
+	[SchemeTagUnused] = { col_gray2, col_gray1, col_gray2 },
+	[SchemeTag1] = { col_cyan, col_gray1, col_gray2 },
+	[SchemeTag2] = { col_red, col_gray1, col_gray2 },
+	[SchemeTag3] = { col_yellow, col_gray1, col_gray2 },
+	[SchemeTag4] = { col_green, col_gray1, col_gray2 },
+	[SchemeTag5] = { col_pink, col_gray1, col_gray2 },
 };
 
 /* tagging */
 static const char *tags[] = { "1", "2", "3", "4", "5", "6", "7", "8", "9" };
+static const int tagschemes[] = { SchemeTag1, SchemeTag2, SchemeTag3, SchemeTag4, SchemeTag5, SchemeTag1, SchemeTag2, SchemeTag3, SchemeTag4 };
 
 static const unsigned int ulinepad	= 5;	/* horizontal padding between the underline and tag */
 static const unsigned int ulinestroke	= 2;	/* thickness / height of the underline */
diff --git a/dwm.c b/dwm.c
index 2a28dec..7f9acfa 100644
--- a/dwm.c
+++ b/dwm.c
@@ -59,7 +59,7 @@
 
 /* enums */
 enum { CurNormal, CurResize, CurMove, CurLast }; /* cursor */
-enum { SchemeNorm, SchemeSel }; /* color schemes */
+enum { SchemeNorm, SchemeSel, SchemeTagUnused, SchemeTag1, SchemeTag2, SchemeTag3, SchemeTag4, SchemeTag5 }; /* color schemes */
 enum { NetSupported, NetWMName, NetWMState, NetWMCheck,
        NetWMFullscreen, NetActiveWindow, NetWMWindowType,
        NetWMWindowTypeDialog, NetClientList, NetClientInfo, NetLast }; /* EWMH atoms */
@@ -867,14 +867,14 @@ drawbar(Monitor *m)
 	x = 0;
 	for (i = 0; i < LENGTH(tags); i++) {
 		w = TEXTW(tags[i]);
-		drw_setscheme(drw, scheme[m->tagset[m->seltags] & 1 << i ? SchemeSel : SchemeNorm]);
+		drw_setscheme(drw, scheme[occ & 1 << i ? tagschemes[i] : SchemeTagUnused]);
 		drw_text(drw, x, 0, w, bh, lrpad / 2, tags[i], urg & 1 << i);
 		if (ulineall || m->tagset[m->seltags] & 1 << i) /* if there are conflicts, just move these lines directly underneath both 'drw_setscheme' and 'drw_text' :) */
 			drw_rect(drw, x + ulinepad, bh - ulinestroke - ulinevoffset, w - (ulinepad * 2), ulinestroke, 1, 0);
-		if (occ & 1 << i)
+		/*if (occ & 1 << i)
 			drw_rect(drw, x + boxs, boxs, boxw, boxw,
 				m == selmon && selmon->sel && selmon->sel->tags & 1 << i,
-				urg & 1 << i);
+				urg & 1 << i);*/
 		x += w;
 	}
 	w = TEXTW(m->ltsymbol);
