diff --git a/config.def.h b/config.def.h
index 43ea6af..94c2a83 100644
--- a/config.def.h
+++ b/config.def.h
@@ -10,11 +10,11 @@ static const int showbar            = 1;        /* 0 means no bar */
 static const int topbar             = 1;        /* 0 means bottom bar */
 static const char *fonts[]          = { "monospace:size=10" };
 static const char dmenufont[]       = "monospace:size=10";
-static const char col_gray1[]       = "#222222";
-static const char col_gray2[]       = "#444444";
-static const char col_gray3[]       = "#bbbbbb";
-static const char col_gray4[]       = "#eeeeee";
-static const char col_cyan[]        = "#005577";
+static const char col_gray1[]       = "#2E3440";
+static const char col_gray2[]       = "#4C566A";
+static const char col_gray3[]       = "#D8DEE9";
+static const char col_gray4[]       = "#ECEFF4";
+static const char col_cyan[]        = "#5E81AC";
 static const char col_red[]         = "#BF616A";
 static const char col_orange[]      = "#D08770";
 static const char col_yellow[]      = "#EBCB8B";
