diff --git a/app/Console/AutoImports.php b/app/Console/AutoImports.php
index c45b296..33b8f00 100644
--- a/app/Console/AutoImports.php
+++ b/app/Console/AutoImports.php
@@ -283,9 +283,6 @@ trait AutoImports
         if (count($this->importErrors) > 0) {
             return 1;
         }
-        if (0 === count($messages) && 0 === count($warnings) && 0 === count($errors)) {
-            return 73;
-        }
 
         return 0;
     }
diff --git a/app/Console/Commands/Import.php b/app/Console/Commands/Import.php
index bc5831b..dff21aa 100644
--- a/app/Console/Commands/Import.php
+++ b/app/Console/Commands/Import.php
@@ -157,9 +157,6 @@ final class Import extends Command
         if (0 !== count($this->importErrors)) {
             $exitCode = 1;
         }
-        if (0 === count($messages) && 0 === count($warnings) && 0 === count($errors)) {
-            $exitCode = 73;
-        }
 
         return $exitCode;
     }
diff --git a/app/Services/Nordigen/Conversion/RoutineManager.php b/app/Services/Nordigen/Conversion/RoutineManager.php
index 047df30..d259cf7 100644
--- a/app/Services/Nordigen/Conversion/RoutineManager.php
+++ b/app/Services/Nordigen/Conversion/RoutineManager.php
@@ -329,7 +329,6 @@ class RoutineManager implements RoutineManagerInterface
         if (0 === $total) {
             app('log')->warning('Downloaded nothing, will return nothing.');
             // add error to current error thing:
-            $this->addError(0, 'No transactions were downloaded from GoCardless. You may be rate limited or something else went wrong.');
             $this->mergeMessages(1);
             $this->mergeWarnings(1);
             $this->mergeErrors(1);
