diff --git a/app/Console/AutoImports.php b/app/Console/AutoImports.php
index 3788afb8..2f85fd21 100644
--- a/app/Console/AutoImports.php
+++ b/app/Console/AutoImports.php
@@ -329,11 +329,6 @@ trait AutoImports
 
             return ExitCode::GENERAL_ERROR->value;
         }
-        if (0 === count($messages) && 0 === count($warnings) && 0 === count($errors)) {
-            Log::error(sprintf('Exit code is %s.', ExitCode::NOTHING_WAS_IMPORTED->name));
-
-            return ExitCode::NOTHING_WAS_IMPORTED->value;
-        }
 
         Log::error(sprintf('Exit code is %s.', ExitCode::SUCCESS->name));
 
diff --git a/app/Console/Commands/Import.php b/app/Console/Commands/Import.php
index 770e376a..f52d17b0 100644
--- a/app/Console/Commands/Import.php
+++ b/app/Console/Commands/Import.php
@@ -121,10 +121,6 @@ final class Import extends Command
             $exitCode = ExitCode::GENERAL_ERROR->value;
             Log::error(sprintf('Exit code is %s.', ExitCode::GENERAL_ERROR->name));
         }
-        if (0 === count($messages) && 0 === count($warnings) && 0 === count($errors)) {
-            $exitCode = ExitCode::NOTHING_WAS_IMPORTED->value;
-            Log::error(sprintf('Exit code is %s.', ExitCode::NOTHING_WAS_IMPORTED->name));
-        }
         if ($exitCode === ExitCode::SUCCESS->value) {
             Log::debug(sprintf('Exit code is %s.', ExitCode::SUCCESS->name));
         }
diff --git a/app/Services/Nordigen/Conversion/RoutineManager.php b/app/Services/Nordigen/Conversion/RoutineManager.php
index 3afd5045..10e0fc22 100644
--- a/app/Services/Nordigen/Conversion/RoutineManager.php
+++ b/app/Services/Nordigen/Conversion/RoutineManager.php
@@ -275,10 +275,6 @@ class RoutineManager implements RoutineManagerInterface
         if (0 === $total) {
             Log::warning('Downloaded nothing, will return nothing.');
             // add error to current error thing:
-            $this->importJob->conversionStatus->addError(
-                0,
-                '[a111]: No transactions were downloaded from GoCardless. You may be rate limited or something else went wrong.'
-            );
             $this->repository->saveToDisk($this->importJob);
 
             return true;
