diff --git a/app/Console/AutoImports.php b/app/Console/AutoImports.php
index 1ab510d..8e9438a 100644
--- a/app/Console/AutoImports.php
+++ b/app/Console/AutoImports.php
@@ -304,11 +304,6 @@ trait AutoImports
 
             return ExitCode::GENERAL_ERROR->value;
         }
-        if (0 === count($messages) && 0 === count($warnings) && 0 === count($errors)) {
-            app('log')->error(sprintf('Exit code is %s.', ExitCode::NOTHING_WAS_IMPORTED->name));
-
-            return ExitCode::NOTHING_WAS_IMPORTED->value;
-        }
 
         app('log')->error(sprintf('Exit code is %s.', ExitCode::SUCCESS->name));
 
diff --git a/app/Console/Commands/Import.php b/app/Console/Commands/Import.php
index b7eddbd..854cf54 100644
--- a/app/Console/Commands/Import.php
+++ b/app/Console/Commands/Import.php
@@ -175,10 +175,6 @@ final class Import extends Command
             $exitCode = ExitCode::GENERAL_ERROR->value;
             app('log')->error(sprintf('Exit code is %s.', ExitCode::GENERAL_ERROR->name));
         }
-        if (0 === count($messages) && 0 === count($warnings) && 0 === count($errors)) {
-            $exitCode = ExitCode::NOTHING_WAS_IMPORTED->value;
-            app('log')->error(sprintf('Exit code is %s.', ExitCode::NOTHING_WAS_IMPORTED->name));
-        }
         if ($exitCode === ExitCode::SUCCESS->value) {
             app('log')->debug(sprintf('Exit code is %s.', ExitCode::SUCCESS->name));
         }
diff --git a/app/Services/Nordigen/Conversion/RoutineManager.php b/app/Services/Nordigen/Conversion/RoutineManager.php
index 4f8069f..d252b5d 100644
--- a/app/Services/Nordigen/Conversion/RoutineManager.php
+++ b/app/Services/Nordigen/Conversion/RoutineManager.php
@@ -328,7 +328,6 @@ class RoutineManager implements RoutineManagerInterface
         if (0 === $total) {
             app('log')->warning('Downloaded nothing, will return nothing.');
             // add error to current error thing:
-            $this->addError(0, '[a111]: No transactions were downloaded from GoCardless. You may be rate limited or something else went wrong.');
             $this->mergeMessages(1);
             $this->mergeWarnings(1);
             $this->mergeErrors(1);
