diff --git a/app/Http/Controllers/Import/Nordigen/LinkController.php b/app/Http/Controllers/Import/Nordigen/LinkController.php
index fa984357..3e6bf5ec 100644
--- a/app/Http/Controllers/Import/Nordigen/LinkController.php
+++ b/app/Http/Controllers/Import/Nordigen/LinkController.php
@@ -103,7 +103,7 @@ class LinkController extends Controller
         $agreementRequest  = new PostNewUserAgreement($url, $accessToken);
         $agreementRequest->setTimeOut(config('importer.connection.timeout'));
         $agreementRequest->setBank($configuration->getNordigenBank());
-        $agreementRequest->setAccessValidForDays('90');
+        $agreementRequest->setAccessValidForDays($configuration->getNordigenAccessDays());
         $agreementRequest->setMaxHistoricalDays($configuration->getNordigenMaxDays());
 
         /** @var NewUserAgreementResponse $agreementResponse */
diff --git a/app/Http/Controllers/Import/Nordigen/SelectionController.php b/app/Http/Controllers/Import/Nordigen/SelectionController.php
index 0fc8ca3d..a1ebc883 100644
--- a/app/Http/Controllers/Import/Nordigen/SelectionController.php
+++ b/app/Http/Controllers/Import/Nordigen/SelectionController.php
@@ -120,9 +120,11 @@ class SelectionController extends Controller
             $values['bank'] = 'SANDBOXFINANCE_SFIN0000';
         }
 
+        $tmp = explode(":", $values['bank'], 2);
         $configuration->setNordigenCountry($values['country']);
-        $configuration->setNordigenBank($values['bank']);
+        $configuration->setNordigenBank($tmp[0]);
         $configuration->setNordigenMaxDays($values['days']);
+        $configuration->setNordigenAccessDays($tmp[1]);
 
         $importJob->setConfiguration($configuration);
         $this->repository->saveToDisk($importJob);
diff --git a/app/Services/Nordigen/Model/Bank.php b/app/Services/Nordigen/Model/Bank.php
index 9419fcd4..0f138e35 100644
--- a/app/Services/Nordigen/Model/Bank.php
+++ b/app/Services/Nordigen/Model/Bank.php
@@ -33,6 +33,7 @@ class Bank
     public string $logo;
     public string $name;
     public int    $transactionTotalDays;
+    public int    $maxAccessValidDays;
 
     /**
      * @return static
@@ -43,6 +44,7 @@ class Bank
         $bank->id                   = $array['id'];
         $bank->name                 = $array['name'];
         $bank->transactionTotalDays = (int) $array['transaction_total_days'];
+        $bank->maxAccessValidDays   = (int) $array['max_access_valid_for_days'];
         $bank->logo                 = $array['logo'];
 
         return $bank;
diff --git a/app/Services/Shared/Configuration/Configuration.php b/app/Services/Shared/Configuration/Configuration.php
index 5dca54cf..1ecc0dd7 100644
--- a/app/Services/Shared/Configuration/Configuration.php
+++ b/app/Services/Shared/Configuration/Configuration.php
@@ -89,6 +89,7 @@ class Configuration
     private string $nordigenBank;
     private string $nordigenCountry;
     private string $nordigenMaxDays;
+    private string $nordigenAccessDays;
     private array  $nordigenRequisitions;
 
     private string $lunchFlowApiKey = '';
@@ -140,6 +141,7 @@ class Configuration
         $this->nordigenBank                = '';
         $this->nordigenRequisitions        = [];
         $this->nordigenMaxDays             = '90';
+        $this->nordigenAccessDays          = '90';
 
         // lunch flow configuration
         $this->lunchFlowApiKey             = '';
@@ -243,6 +245,7 @@ class Configuration
         $object->nordigenBank                = $data['nordigen_bank'] ?? '';
         $object->nordigenRequisitions        = $data['nordigen_requisitions'] ?? [];
         $object->nordigenMaxDays             = $data['nordigen_max_days'] ?? '90';
+        $object->nordigenAccessDays          = $data['nordigen_access_days'] ?? '90';
 
         // lunch flow configuration
         $object->lunchFlowApiKey             = $data['lunch_flow_api_key'] ?? '';
@@ -396,6 +399,7 @@ class Configuration
         $object->nordigenBank                = $array['nordigen_bank'] ?? '';
         $object->nordigenRequisitions        = $array['nordigen_requisitions'] ?? [];
         $object->nordigenMaxDays             = $array['nordigen_max_days'] ?? '90';
+        $object->nordigenAccessDays          = $array['nordigen_access_days'] ?? '90';
 
         // lunch flow configuration
         $object->lunchFlowApiKey             = $array['lunch_flow_api_key'] ?? '';
@@ -494,6 +498,7 @@ class Configuration
         $object->nordigenBank                = $array['nordigen_bank'] ?? '';
         $object->nordigenRequisitions        = $array['nordigen_requisitions'] ?? [];
         $object->nordigenMaxDays             = $array['nordigen_max_days'] ?? '90';
+        $object->nordigenAccessDays          = $array['nordigen_access_days'] ?? '90';
 
         // lunch flow configuration
         $object->lunchFlowApiKey             = $array['lunch_flow_api_key'] ?? '';
@@ -760,6 +765,16 @@ class Configuration
         $this->nordigenMaxDays = $nordigenMaxDays;
     }
 
+    public function getNordigenAccessDays(): string
+    {
+        return $this->nordigenAccessDays;
+    }
+
+    public function setNordigenAccessDays(string $nordigenAccessDays): void
+    {
+        $this->nordigenAccessDays = $nordigenAccessDays;
+    }
+
     public function getNordigenRequisitions(): array
     {
         return $this->nordigenRequisitions;
@@ -985,6 +1000,7 @@ class Configuration
             'nordigen_bank'                => $this->nordigenBank,
             'nordigen_requisitions'        => $this->nordigenRequisitions,
             'nordigen_max_days'            => $this->nordigenMaxDays,
+            'nordigen_access_days'         => $this->nordigenAccessDays,
             'lunch_flow_api_key'           => $this->lunchFlowApiKey,
 
             // utf8
diff --git a/resources/views/v2/import/009-selection/index.blade.php b/resources/views/v2/import/009-selection/index.blade.php
index e87bc2c5..fca1dc2b 100644
--- a/resources/views/v2/import/009-selection/index.blade.php
+++ b/resources/views/v2/import/009-selection/index.blade.php
@@ -94,7 +94,7 @@
                                         <option label="{{ $bank->name }}"
                                                 @if($bank->id === $configuration->getNordigenBank())selected="selected"@endif
                                                 data-days="{{ $bank->transactionTotalDays }}"
-                                                value="{{ $bank->id }}">{{ $bank->name }}</option>
+                                                value="{{ $bank->id }}:{{ $bank->maxAccessValidDays }}">{{ $bank->name }}</option>
                                         @endforeach
                                     </select>
                                 </div>
