diff --git a/app/Console/AutoImports.php b/app/Console/AutoImports.php
index 3c82494f..7581dfc7 100644
--- a/app/Console/AutoImports.php
+++ b/app/Console/AutoImports.php
@@ -256,7 +256,8 @@ trait AutoImports
                     array_merge($this->conversionMessages, $this->importMessages),
                     array_merge($this->conversionWarnings, $this->importWarnings),
                     array_merge($this->conversionErrors, $this->importErrors),
-                    $this->conversionRateLimits
+                    $this->conversionRateLimits,
+                    $file
                 )
             );
 
@@ -274,7 +275,7 @@ trait AutoImports
         $messages      = array_merge($this->importMessages, $this->conversionMessages);
         $warnings      = array_merge($this->importWarnings, $this->conversionWarnings);
         $errors        = array_merge($this->importErrors, $this->conversionErrors);
-        event(new ImportedTransactions($messages, $warnings, $errors, $this->conversionRateLimits));
+        event(new ImportedTransactions($messages, $warnings, $errors, $this->conversionRateLimits, $file));
 
         if (count($this->importErrors) > 0 || count($this->conversionRateLimits) > 0) {
             app('log')->error(sprintf('Exit code is %s.', ExitCode::GENERAL_ERROR->name));
@@ -643,7 +644,8 @@ trait AutoImports
                 array_merge($this->conversionMessages, $this->importMessages),
                 array_merge($this->conversionWarnings, $this->importWarnings),
                 array_merge($this->conversionErrors, $this->importErrors),
-                $this->conversionRateLimits
+                $this->conversionRateLimits,
+                $jsonFile
             )
         );
     }
diff --git a/app/Events/ImportedTransactions.php b/app/Events/ImportedTransactions.php
index 6e1dbb83..f8bef42b 100644
--- a/app/Events/ImportedTransactions.php
+++ b/app/Events/ImportedTransactions.php
@@ -40,8 +40,9 @@ class ImportedTransactions
     public array $messages;
     public array $warnings;
     public array $rateLimits;
+    public string $fileName;
 
-    public function __construct(array $messages, array $warnings, array $errors, array $rateLimits)
+    public function __construct(array $messages, array $warnings, array $errors, array $rateLimits, string $fileName = null)
     {
         app('log')->debug('Created event ImportedTransactions with filtering (2)');
 
@@ -50,6 +51,7 @@ class ImportedTransactions
         $this->warnings   = $this->filterArray('warning(s)', $warnings);
         $this->errors     = $this->filterArray('error(s)', $errors);
         $this->rateLimits = $this->filterArray('rate limit message(s)', $rateLimits);
+        $this->fileName   = $fileName;
     }
 
     /**
diff --git a/app/Handlers/Events/ImportedTransactionsEventHandler.php b/app/Handlers/Events/ImportedTransactionsEventHandler.php
index b4205c8b..d9430907 100644
--- a/app/Handlers/Events/ImportedTransactionsEventHandler.php
+++ b/app/Handlers/Events/ImportedTransactionsEventHandler.php
@@ -59,6 +59,7 @@ class ImportedTransactionsEventHandler
             'warnings'    => $event->warnings,
             'errors'      => $event->errors,
             'rate_limits' => $event->rateLimits,
+            'fileName'    => $event->fileName,
         ];
         if (count($event->messages) > 0 || count($event->warnings) > 0 || count($event->errors) > 0) {
             app('log')->info('Will send report message.');
diff --git a/app/Mail/ImportReportMail.php b/app/Mail/ImportReportMail.php
index a0e1dcf8..3db93332 100644
--- a/app/Mail/ImportReportMail.php
+++ b/app/Mail/ImportReportMail.php
@@ -44,6 +44,7 @@ class ImportReportMail extends Mailable
     public string $version;
     public array  $warnings;
     public array  $rateLimits;
+    public string $fileName;
 
     /**
      * Create a new message instance.
@@ -60,6 +61,7 @@ class ImportReportMail extends Mailable
         $this->warnings   = $log['warnings'] ?? [];
         $this->messages   = $log['messages'] ?? [];
         $this->rateLimits = $log['rate_limits'] ?? [];
+        $this->fileName   = $log['fileName'] ?? [];
     }
 
     /**
diff --git a/resources/views/v2/emails/import/report.blade.php b/resources/views/v2/emails/import/report.blade.php
index 4d0128d1..fc4f86b3 100644
--- a/resources/views/v2/emails/import/report.blade.php
+++ b/resources/views/v2/emails/import/report.blade.php
@@ -58,5 +58,6 @@ Go to Firefly III
 
 Enjoy!<br>
 Firefly III Data Importer, @if(str_starts_with($version, 'develop')){{ $version }} @else v{{ $version }}@endif
+@if('' !== $fileName)<br>{{ $fileName }}@endif
 
 @endcomponent
