diff --git a/app/Services/Nordigen/Model/Transaction.php b/app/Services/Nordigen/Model/Transaction.php
index 3c07673b..81c66691 100644
--- a/app/Services/Nordigen/Model/Transaction.php
+++ b/app/Services/Nordigen/Model/Transaction.php
@@ -150,7 +150,7 @@ class Transaction
         $object->endToEndId                             = trim($array['endToEndId'] ?? ''); // from Rabobank NL
 
         // overrule transaction id when empty using the internal ID:
-        if ('' === $object->transactionId) {
+        if ('' === $object->transactionId || str_starts_with($object->transactionId, "FOBA")) {
             $object->transactionId = trim($array['internalTransactionId'] ?? '');
         }
 
