diff --git a/app/Console/AutoImports.php b/app/Console/AutoImports.php
index 2f85fd21..dbe64d1e 100644
--- a/app/Console/AutoImports.php
+++ b/app/Console/AutoImports.php
@@ -264,9 +264,9 @@ trait AutoImports
             Log::warning('Almost tried to import a JSON file as a file lol. Skip it.');
 
             // don't report this.
-            Log::debug(sprintf('[%s] Exit code is %s.', config('importer.version'), ExitCode::GENERAL_ERROR->name));
+            Log::debug(sprintf('[%s] Exit code is %s.', config('importer.version'), ExitCode::SUCCESS->name));
 
-            return ExitCode::GENERAL_ERROR->value;
+            return ExitCode::SUCCESS->value;
         }
 
         $this->line(sprintf('[a] Going to convert from file "%s" using configuration %s and flow "%s".', $importableFile, $jsonFile, $configuration->getFlow()));
