diff --git a/app/Http/Controllers/IndexController.php b/app/Http/Controllers/IndexController.php
index daff31c..c213746 100644
--- a/app/Http/Controllers/IndexController.php
+++ b/app/Http/Controllers/IndexController.php
@@ -53,7 +53,6 @@ class IndexController extends Controller
             cookie(Constants::FLOW_COOKIE, ''),
         ];
         \Artisan::call('cache:clear');
-        \Artisan::call('config:clear');
 
         return redirect(route('index'))->withCookies($cookies);
     }
