diff --git a/src/main.rs b/src/main.rs
index 0a8d2936284375..988b89ad5bf617 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -1170,6 +1170,7 @@ won't take effect until you reboot the system.
             move |signal: OrgFreedesktopSystemd1ManagerReloading,
                   _: &LocalConnection,
                   _msg: &Message| {
+                log::debug!("reload status: {}", &signal.active);
                 *_systemd_is_reloading.borrow_mut() = signal.active;
 
                 true
@@ -1708,6 +1709,7 @@ won't take effect until you reboot the system.
         .context("Failed to reset failed units")?;
 
     // Make systemd reload its units.
+    log::debug!("old reload status: {}", &systemd_is_reloading.borrow());
     _ = systemd.reload(); // we don't get a dbus reply here
     *systemd_is_reloading.borrow_mut() = true;
     log::debug!("waiting for systemd reload to finish");
@@ -1717,6 +1719,7 @@ won't take effect until you reboot the system.
             .process(Duration::from_millis(500))
             .context("Failed to process dbus messages")?;
     }
+    log::debug!("systemd has reloaded");
 
     dbus_conn
         .remove_match(reloading_token)
